include/walnuts/walnuts.hpp

```diff
@@ -497,17 +496,22 @@ class WalnutsSampler {
    * @brief Construct a WALNUTS sampler from the specified RNG, target log
    * density/gradient initialization, and tuning parameters.
    *
-   * @param rand The randomizer for HMC.
-   * @param logp_grad The target log density and gradient function (see the
-   * class documentation.
+   * The template parameter `U` allows this function to either wrap a
+   * bare `logp_grad` function or accept a `LogpGrad` instance to
+   * copy (it will not double wrap).
+   *
+   * @tparam U The type of the target log density/gradient function.
+   * @param rand The compound randomizer for HMC.
+   * @param logp_grad The target log density/gradient function.
```

Comment:
If you really want to preserve the log density count between the two, I think a more intuitive way to do this is to define a second constructor with the signature
`WalnutsSampler(Random<S, RNG> &rand, const LogpGrad<F> &logp_grad, const Vec<S> &theta, ... `

I think this would let you recover CTAD, at least

Reply:
That's how I started.  Then I noticed it was an exact duplicate of the code and doc with the only difference being the type `LogpGrad<F>` vs. `F`.

Reply:
I'm just going to remove this counter altogether in the interest of moving on with development.